packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useQuery, useLazyQuery, useMutation } from '@apollo/react-hooks';
+
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+
+export const displayStates = {
+    DONE: 'done',
+    EDITING: 'editing'
+};
+
+export const useShippingMethod = props => {
+    const {
+        onSave,
+        queries: { getShippingMethods, getSelectedShippingMethod },
+        mutations: { setShippingMethod }
+    } = props;
+
+    const [{ cartId }] = useCartContext();
+
+    /*
+     *  Apollo Hooks.
+     */
+    const {
+        data: chosenShippingMethodData,
+        loading: isLoadingSelectedShippingMethod
+    } = useQuery(getSelectedShippingMethod);
```

Comment:
👻 BOO!

😄 Resolved by https://github.com/magento/pwa-studio/pull/2280/commits/f142a6b178ece53e7a577f1096002d7c9e8aabb8

And nice catch actually, I'm not sure how the `useQuery` was working before because I never passed it a `cartId` `variable`...

Reply:
Spooky!